lib/routes/account.js

```diff
@@ -341,23 +362,43 @@ module.exports = function (
         var email = form.email
         var authPW = Buffer(form.authPW, 'hex')
         var service = request.payload.service || request.query.service
-        var emailRecord, sessionToken, device
+        var redirectTo = request.payload.redirectTo
+        var resume = request.payload.resume
+        var tokenVerificationId = crypto.randomBytes(16)
+        var emailRecord, sessionToken, device, keyFetchToken, doSigninConfirmation
 
         customs.check(request.app.clientAddress, email, 'accountLogin')
           .then(readEmailRecord)
           .then(createSessionToken)
+          .then(createKeyFetchToken)
           .then(upsertDevice)
           .then(emitSyncLoginEvent)
-          .then(sendNewDeviceLoginNotification)
+          .then(sendVerifyLoginEmail)
           .then(createResponse)
           .done(reply, reply)
 
-        function readEmailRecord (result) {
+        function shouldEnableSigninConfirmation(uid) {
+          // Sign-in confirmation is determined by whether the feature is enabled and
+          // if the user's first two uid digits fall within the config sample rate.
+          if (config.signinConfirmation.enabled) {
+
+            // Returns the first 2 integers of uid
+            var uidNum = uid.match(/\d/g).splice(0, 2).join('')
```

Comment:
It's possible, however unlikely, that the uid does not contain any integers.  It may be a little safer to do something like:

```
uidNum = parseInt(uid.substr(0, 4), 16) % 100
```